### PR TITLE
feat: support sending events when app move to background

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
## Description
1. add background task for sending request when app move to background.
2. change event log to prettier format.

## General Checklist

- [x] Added new tests to cover change, if needed
- [x] Build succeeds using Swift Package Manager
- [x] All unit tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift [hunk 1]
```diff
@@ -15,8 +18,8 @@ protocol AnalyticsEventRecording {
     func save(_ event: ClickstreamEvent) throws
 
     /// Submit locally stored events
-    /// - Returns: A collection of events submitted to Clickstream
-    func submitEvents()
+    /// - Parameter inBackgroundMode: wheter use background mode to send request
```
Comment (zxkane) [inline Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift:21]: Typo -> Whether using

File: Sources/Clickstream/Dependency/Clickstream/Analytics/AnalyticsClient.swift [hunk 2]
```diff
@@ -124,7 +124,7 @@ class AnalyticsClient: AnalyticsClientBehaviour {
         try eventRecorder.save(event)
     }
 
-    func submitEvents() {
-        eventRecorder.submitEvents()
+    func submitEvents(inBackgroundMode: Bool = false) {
```
Comment (zxkane) [inline Sources/Clickstream/Dependency/Clickstream/Analytics/AnalyticsClient.swift:127]: Is parameter isBackgroundMode better?

File: Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift [hunk 3]
```diff
@@ -61,10 +63,20 @@ class EventRecorder: AnalyticsEventRecording {
...
         if queue.operationCount < Constants.maxEventOperations {
             let operation = BlockOperation { [weak self] in
-                _ = self?.processEvent()
+                if inBackgroundMode {
+                    #if canImport(UIKit)
+                        let taskId = UIApplication.shared.beginBackgroundTask(expirationHandler: nil)
+                        self?.log.debug("Start background task")
+                        _ = self?.processEvent()
+                        UIApplication.shared.endBackgroundTask(taskId)
+                        self?.log.debug("Ended background task")
```
Comment (zxkane) [inline Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift:75]: ‘Background task is the end’ is better

File: Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift [hunk 4]
```diff
@@ -133,6 +144,15 @@ class EventRecorder: AnalyticsEventRecording {
         }
         return BatchEvent(eventsJson: eventsJson, eventCount: eventCount, lastEventId: lastEventId)
     }
+
+    func logEventPrettier(event: ClickstreamEvent) {
+        log.debug("Event saved, event name: \(event.eventType)")
+        print("attributes: {")
```
Comment (zxkane) [inline Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift:150]: Why using print here

**Review by zxkane — CHANGES_REQUESTED** [review]

File: Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift (hunk 1 above)
Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift:21]: fixed.

File: Sources/Clickstream/Dependency/Clickstream/Analytics/AnalyticsClient.swift (hunk 2 above)
Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/Analytics/AnalyticsClient.swift:127]: Yes, changed to `isBackgroundMode`

File: Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift (hunk 3 above)
Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift:75]: Changed to "Background task is the end"

File: Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift (hunk 4 above)
Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift:150]: `log.debug()` function will log the default time and class path information, when the attribute value length is long, for example event `product_exposure`, the log will looks unfriendly:
image
when use `print(xxx)`can directly print the content in output：
image
Analytics SDK event logs are generally printed as above.

Comment (zxkane) [reply Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift:150]: Using the system's print directly will result in the loss of the capability to overwrite the configuration of the logger, which is not something I favor.

Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift]: I also think it's not a good way. Fixed it for using common log util and also have prettier output.

**Review by zxkane — APPROVED** [review]

**Pull request merged**